hack/update-deps.sh

```diff
@@ -48,9 +47,21 @@ readonly GO_GET
 
 if (( GO_GET )); then
   go get -d ${FLOATING_DEPS[@]}
+
+  # Update test-infra
+  # TODO(coryrc): move this to a file in knative/test-infra/scripts
```

Comment:
Can we have that added to `build.sh -u`, too, as this is the main build tool (we should move update-deps.sh to build.sh, and add another option for updating the dependencies)

Reply:
Thinking again, we should probably do it in a latter step, but having kind of  a way to specify the version of test-infra to update to might make sense, too.

Reply:
@coryrc Adding the update to this script actually contradicts your reasoning in https://github.com/knative/test-infra/issues/2069 as it will update go.mod and the test-infra script at the same pace, and not independently. Its better to call it from externally, like from `hack/build.sh`

Reply:
It's run from `hack/update-deps.sh --upgrade` because we need it to upgrade when the user intends, but if the test-infra oncall needs to update scripts they would just run `scripts/test-infra/update-test-infra.sh --update`.

Reply:
Actually, I could do that in this PR, hold on...

Reply:
Oh, I see now there isn't a `build.sh -u`

I think you will want to keep `hack/update-deps.sh` because it's what the auto-update jobs call to create PRs to update vendors automatically; even if you make `build.sh -u` just call `update-deps.sh --upgrade`.

Reply:
or vice versa ;-) 

I agree that we should keep the hooks (== scripts) that are called from the outside. However it is not often clear which scripts those are, and I don't know of any documentation that has comprehensive lists of hooks into a repository (but this could be also only my ignorance)